test/functional/Controller/ApiControllerTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OnMoon\OpenApiServerBundle\Test\Functional\Controller;
+
+use OnMoon\OpenApiServerBundle\Command\GenerateApiCodeCommand;
+use OnMoon\OpenApiServerBundle\Controller\ApiController;
+use OnMoon\OpenApiServerBundle\Interfaces\ApiLoader;
+use OnMoon\OpenApiServerBundle\Test\Functional\TestKernel;
+use PHPUnit\Framework\Assert;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\BrowserKit\HttpBrowser;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\Response;
+
+use function Safe\file_put_contents;
+
+/**
+ * @covers \OnMoon\OpenApiServerBundle\Controller\ApiController
+ */
+class ApiControllerTest extends WebTestCase
+{
+    /**
+     * phpcs:disable SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingNativeTypeHint
+     * @var string $class
+     */
+    protected static $class = TestKernel::class;
+
+    public function testGetApiReturnsOkRequest(): void
+    {
+        /** @var HttpBrowser $client */
+        $client        = static::createClient();
+        $application   = new Application(static::$kernel);
+        $command       = $application->find(GenerateApiCodeCommand::COMMAND);
+        $commandTester = new CommandTester($command);
+        $commandTester->execute(['command' => GenerateApiCodeCommand::COMMAND]);
+        $getGoodImplClassName = $this->createGetGoodImpl();
+
+        static::$container->set('petstore.getGood', new $getGoodImplClassName());
+
+        $apiLoaderClass = TestKernel::$bundleRootNamespace . '\ServiceSubscriber\ApiServiceLoaderServiceSubscriber';
+        /** @var ApiLoader $apiLoader */
+        $apiLoader = new $apiLoaderClass(static::$container);
+
+        /** @var ApiController $apiController */
+        $apiController = static::$container->get(ApiController::class);
+        $apiController->setApiLoader($apiLoader);
```

Comment:
Could this setup be moved to the kernel class? It's ok if there will be multiple specific kernel classes for each test

Reply:
There is a trouble with container. We can't run `generate` command from kernel so we couldn't set a `petstore.getGood` service to the container because DTO interface hasn't been generated yet.